### PR TITLE
Removed the distinction of boards from other groups in fetching and handling data

### PR DESCRIPTION
**Pull request by Scarletto**
There is no internal functional difference at all between boards and other groups.
The only reason `BoardCubit` and `BoardScreen` were separate is because fetching boards from the API worked by year slug as opposed to PK.

However, boards do in fact have a PK that is pulled from the same pile as the PKs of committees and societies, and it can be used to fetch board data at the same API endpoint as that of committees and societies.
In fact, the year slug previously used to fetch board data is not even part of the board data the API responds with. This means the only way to construct the slug from e.g. a `ListGroup` of a board (in order to do an API call for the board page itself) would be a hacky way like string manipulation on a board name.

Since this is quite annoying to work with, and the API is already primed for fetching boards like any other group anyway, I don't see a reason to keep these separate (and hence I removed the distinction in this PR).

### Summary
Removed `BoardCubit`, `BoardScreen`.
~~Changed routing for boards to construct a `GroupScreen` rather than a `BoardScreen` with a groupPk rather than a boardSlug.~~

EDIT:
Added a `.bySlug()` constructor to `GroupCubit` and `GroupScreen` and changed fields to support either PK or `MemberGroupType` + slug.
Added route `board` for creation by PK; changed old `board` (by slug) to `boardBySlug`.

EDIT 2:
Changed `getBoardGroup` to `getGroupBySlug` and implemented every call generically (so passing `MemberGroupType` + slug).

### CODE REVIEW

File: lib/blocs/group_cubit.dart [hunk 1]
```diff
@@ -29,25 +29,3 @@ class GroupCubit extends Cubit<GroupState> implements BaseGroupCubit {
     }
   }
 }
```
Comment (JAicewizard) [inline lib/blocs/group_cubit.dart:31]: `GroupCubit` can now be merged with the `BaseGroupCubit`

Comment (Scarletto) [reply lib/blocs/group_cubit.dart:26]: Nice, it has been done!

**Comment by DeD1rk** [comment]
Not having this strange distinction would be nice. However, there was of course a reason to have it in the first place... without it, you cant open a bord by its deeplink.

**Comment by Scarletto** [comment]
That makes sense, but it still seems really odd to change the class specifically for this.
Can I just add a route that gets the PK from the slug, then creates the GroupScreen?
Really the only other alternative is readding the class and then adding another class specifically for constructing with PK instead of slug.

**Comment by DeD1rk** [comment]
That would be fine I think. Will just make you end up with yet another cubit I guess, but as you kind of need the pk-based thing too that seems inevitable

File: lib/ui/screens/group_screen.dart [hunk 2]
```diff
@@ -10,48 +10,49 @@ import 'package:reaxit/ui/widgets.dart';
...
+            pk: pk!,
+          )..load();
+        } else {
+          return GroupCubit.bySlug(
+            RepositoryProvider.of<ApiRepository>(context),
+            groupType: groupType!,
+            slug: slug!,
+          )..load();
+        }
+      },
```
Comment (JAicewizard) [inline lib/ui/screens/group_screen.dart:51]: This could be a seperate function for readability, like `_selectCubit`

Comment (Scarletto) [reply lib/ui/screens/group_screen.dart]: Donezo!

**Review by JAicewizard — APPROVED** [review]

**Pull request merged**